components/director/internal/tenantfetchersvc/handler.go

```diff
@@ -109,16 +115,73 @@ func (h *handler) DeleteByExternalID(writer http.ResponseWriter, req *http.Reque
 	writer.WriteHeader(http.StatusOK)
 }
 
-func (h *handler) tenantInfoFromBody(body []byte) (*model.BusinessTenantMappingInput, error) {
+func (h *handler) CreateRegional(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	vars := mux.Vars(request)
+	region, ok := vars[h.config.RegionPathParam]
+	if !ok {
+		log.C(ctx).Errorf("Region path parameter is missing from request")
+		http.Error(writer, "Region path parameter is missing from request", http.StatusBadRequest)
+		return
+	}
+
+	body, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("Failed to read tenant information from request body: %v", err)
+		http.Error(writer, "Failed to read tenant information from request body", http.StatusInternalServerError)
+		return
+	}
+	regionalTenant, err := h.tenantFromBody(body)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("Failed to extract tenant information from request body: %v", err)
+		http.Error(writer, fmt.Sprintf("Failed to extract tenant information from request body: %s", err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	if err := h.provisionTenant(ctx, *regionalTenant, region); err != nil {
+		log.C(ctx).WithError(err).Errorf("Failed to provision tenant with ID %s: %v", regionalTenant.ExternalTenant, err)
+		http.Error(writer, fmt.Sprintf(tenantCreationFailureMsgFmt, regionalTenant.ExternalTenant), http.StatusInternalServerError)
+		return
+	}
+
+	writer.Header().Set("Content-Type", "text/plain")
+	writer.WriteHeader(http.StatusOK)
+	if _, err := writer.Write([]byte(compassURL)); err != nil {
+		log.C(ctx).WithError(err).Errorf("Failed to write response body for tenant request creation for tenant %s: %v", regionalTenant.ExternalTenant, err)
+	}
```

Comment:
This functionality is the same as in the `Create` method. Extract it to a common method